packages/data-stores/src/woocommerce-eligibility/selectors.ts

```diff
@@ -0,0 +1,102 @@
+import { select } from '@wordpress/data';
+import { get } from 'lodash';
+import { STORE_KEY, statusMapping } from './constants';
+import { TransferEligibilityWarning, TransferEligibilityError, TransferEligibility } from './types';
+import type { State } from './reducer';
+
+export const getState = ( state: State ) => state;
+
+export const getAutomatedTransferEligibility = (
+	state: State,
+	siteId: number | null
+): TransferEligibility | null => {
+	if ( ! state || ! state.transferEligibility ) {
+		return null;
+	}
+
+	if ( ! siteId ) {
+		return null;
+	}
+	return state.transferEligibility[ siteId ];
+};
+
+export const getEligibilityHolds = (
+	state: State,
+	siteId: number | null
+): TransferEligibilityError[] | null => {
+	const transferEligibility = select( STORE_KEY ).getAutomatedTransferEligibility( siteId );
```

Comment:
Selectors are supposed to read only from `state`, never from anywhere else:
```js
const transferEligibility = getAutomatedTransferEligibility( state, siteId );
```

Reply:
I may have misunderstood but I thought it was necessary to do `select( STORE_KEY ).getAutomatedTransferEligibility( siteId )` to ensure that the corresponding resolver was called in the event the state hadn't been populated yet.

@alshakero Can you let me know if I'm mistaken about that?

Reply:
I think you're right. Maybe @jsnajdr didn't know this selector has a resolver?

Calling `getAutomatedTransferEligibility` directly won't trigger the resolver because the store has no way to know you called `getAutomatedTransferEligibility` to go ahead and trigger the resolver.

But when you do `select( STORE_KEY ).getAutomatedTransferEligibility( siteId )` you go through the store, giving it a chance to do whatever it needs to do (e.g., trigger the resolver).

I think this is one of the two ways to chain selectors, this and [createRegistrySelector](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#createregistryselector).

Reply:
Splendid, thanks for clarifying!

Reply:
Oh, I see, you need to trigger the resolver. Then you can either use `createRegistrySelector`, or create a "forwarding" resolver for every selector that depends on the main one:
```js
const derivedSelector = ( state ) => originalSelector( state ) * 2;
const derivedResolver = ( /* args */ ) => ( { resolveSelect } ) => resolveSelect.originalSelector( /* args */ );
```

The trouble with global `select( STORE_KEY )` is that it always refers to the instance registered in the default registry. But you can create many registries where each has its own instance and in that case these selectors wouldn't work. It's like not using `this.foo()` on an object instance and using a global shared variable instead.

Reply:
Oh great explanation. This is even more relevant in data-stores package because it's meant to be as reusable as possible. I'd go with `createRegistrySelector`. It's clean and minimalistic.

/cc @razvanpapadopol I remember calling global `select` was your solution, you might want to know this.